code/game/machinery/computer/ftl_weapons.dm

```diff
@@ -196,7 +194,10 @@
 
 			. = 1
 		if("toggle_shields")
-			SSstarmap.ftl_shieldgen.on = !SSstarmap.ftl_shieldgen.on
+			if(!SSstarmap.ftl_shieldgen.stat)
+				SSstarmap.ftl_shieldgen.stat |= POWEROFF
+			else if(SSstarmap.ftl_shieldgen.stat)
+				SSstarmap.ftl_shieldgen.stat &= ~POWEROFF
```

Comment:
I just realized you could skip the if else with a XOR
`SSstarmap.ftl_shieldgen.stat ^= POWEROFF`